janus/lib/client/jsx/project-view.jsx

```diff
@@ -0,0 +1,191 @@
+        editable
+          ? <input className='affiliation' type='text' name='affiliation' value={ revision.affiliation || affiliation || ''}
+              onChange={ (e) => update( 'affiliation', e.target.value ) } />
+          : <span className='affiliation'>{ affiliation }</span>
+      }
+    </div>
+    <div className='cell'>
+      {
+        editable && !create
+          ? <input type='checkbox' name='privileged' checked={ 'privileged' in revision ? revision.privileged : privileged } onChange={ e => update('privileged', e.target.checked) } />
+          : (privileged && 'Yes')
+      }
+    </div>
+    <div className='cell submit'>
+      {
+        Object.keys(revision).length > 0 && <React.Fragment>
+          <Icon className='approve' icon='save' onClick={ () => { onSave({revision,user_email}); updateRevision({}); } } />
+          <Icon className='cancel' icon='ban' onClick={ () => updateRevision({}) }/>
+        </React.Fragment>
+      }
+    </div>
+  </div>
+}
+
+const userCount = (count, txt) => count == 0 ? null : `${count} ${txt}${ count === 1 ? '' : 's' }`;
+const ProjectView = ({project_name}) => {
+  let user = useReduxState( state => selectUser(state) );
+  let [ project, setProject ] = useState({});
+  let [ filter, setFilter ] = useState(null);
+
+  let retrieveProject = useCallback(
+    () => {
+      json_get(`/project/${project_name}`)
+        .then(({project}) => setProject(project))
+    }, [project_name, setProject]
+  );
+  useEffect( () => retrieveProject(), [] );
+
+  let { permissions=[], project_name_full } = project;
+
+  let privileged = permissions.filter(p => p.privileged);
+
+  let editable = isAdmin(user, project_name);
+  let roles = isSuperuser(user,project_name)
+    ?  [ 'administrator', 'editor', 'viewer', 'disabled' ]
+    : isAdmin(user, project_name)
+      ? [ 'editor', 'viewer', 'disabled' ]
+      : [ ];
+
+  return <div id='project-view'>
+    <div className='title'> { project_name_full }</div>
+    <div className='item summary'>
+      {
+        [ 'administrator', 'editor', 'viewer' ].map(
+          role => {
+            let ps = permissions.filter(p => p.role == role);
+            return userCount(ps.length, role)
+          }
+        ).concat(
+          userCount(privileged.length, 'privileged user')
+        ).filter(_=>_).join(', ')
+      }
+    </div>
+    <div className='item header'>
+      <div className='cell'>Name</div>
+      <div className='cell'>Email</div>
+      <div className='cell'>Role</div>
+      <div className='cell'>Affiliation</div>
+      <div className='cell'>Privileged</div>
+      <div className='cell submit'></div>
+    </div>
+    {
+      permissions.length > 10 && <div className='item'>
+        <input className='filter' type='text' placeholder='Filter rows' name='filter' onChange={ e => setFilter(e.target.value) }/>
+      </div>
+    }
+    {
+      permissions.sort((a,b) => (a.role+a.user_email).localeCompare(b.role+b.user_email) ).filter(
```

Comment:
might make sense to to the sort / localeCompare up closer to the deconstruction rather than down here.  Will make the view code a little cleaner.